### DISCUSSION TITLE
Dreame D9 crashing when starting a clean

### DISCUSSION THREAD

**Discussion by mals** [Hypfer/Valetudo#1556] [Q&A / Support] [answered]
The vacuum appears to be crashing within a few seconds of starting a clean. What I am observing is that the vacuum stops working and the robot stops moving. I am no longer able to ping the vacuum or ssh to it. This happens within the first 30 seconds of starting a clean but usually within the first 5 seconds.

## To Reproduce
Press the power button to start a clean (or start a clean through the Valetudo UI)

### Vacuum Model
Dreame D9

### Valetudo Version
2022.08.0
Firmware 1132

## Expected behavior
The vacuum should not crash and continue cleaning

## Additional context
This is an initial installation of Valetudo on a newly rooted Dreame D9. I am attempting to downgrade to FW 1100 with Valetudo 2022.08.0.

**Comment by mals** [comment]
Downgrading to FW 1100 appears to have fixed the problem.

**Reply by MonsterEnergy-wtf** [reply]
Stupid question: how did you achieve this?
my Dreame D9 says that it cannot open /etc/os_release: no such file blahblah. No chance to get into the shell there.
Any idea?

**Comment by Hypfer** [comment, marked as answer]
This issue is caused by the MCU Firmware being too old.
Note that there is no way to display the currently running MCU firmware version from the linux firmware.

To fix this, try the following:
- Make sure that the robot is docked
- Download the dustbuilder firmware update package onto the robot
- run install-manual.sh
- Wait for it to reboot
- Download the dustbuilder firmware update package onto the robot
- run install.sh
- Wait for it to reboot
- Download the dustbuilder firmware update package onto the robot
- run install-mcufw.sh
- Wait for it to reboot

If it is not yet fixed, try repeating the install-mcufw.sh step a few times

**Reply by mals** [reply]
Thanks, this fixed it!

**Reply by baurst** [reply]
I had the same issue with my Dreame D9 Pro (bought in October 2022). This guide (esp. step `run install-mcufw.sh`) solved it. Thank you!

**Reply by Syrk08** [reply]
Hello,
I've having the same problem on my D9.. How am I supposed to do that?! :[
Do i need to some kind of root it?! 

Thanks
